LearningMissionSimulation/PlaygroundEdvard.cs

```diff
@@ -316,22 +366,54 @@ public void SimulationBig1(int subjectCount, int moduleCount)
         #region Reports
         void ReportHeader(string actionName)
         {
-            Console.WriteLine($"******{actionName} is started******\n");
+            switch (this.ReportType)
+            {
+                case ReportType.Verbose:
+                case ReportType.Short:
+                case ReportType.Error:
+                    Console.WriteLine($"******{actionName} is started******\n");
+                    break;
+            }
         }
 
         void ReportFooter(string actionName)
         {
-            Console.WriteLine($"******{actionName} is finished******\n");
+            switch (this.ReportType)
+            {
+                case ReportType.Verbose:
+                case ReportType.Short:
+                case ReportType.Error:
+                    Console.WriteLine($"******{actionName} is finished******\n");
+                    break;
+            }
         }
 
         void ReportItem(string itemName, string actionName, int itemIndex)
         {
-            Console.WriteLine($" {actionName} {itemIndex}\n{itemName}");
+            if (this.ReportType == ReportType.Verbose)
+            {
+                Console.WriteLine($" {actionName} {itemIndex} {itemName}\n");
+            }
         }
 
         void ReportSummary(string summary, int count)
         {
-            Console.WriteLine($"''''''Generated {count} {summary}''''''\n");
+            if (this.ReportType == ReportType.Verbose || this.ReportType == ReportType.Short)
+            {
+                Console.WriteLine($"''''''Generated {count} {summary}''''''\n");
```

Comment:
using "Generated" is not working well for all summaries calls, consider using actionName parameter instead of the summary parameter and update message in line 403 accordingly.

Reply:
Done.